frappe/public/js/legacy/clientscriptAPI.js

```diff
@@ -66,8 +66,19 @@ refresh_field = function(n, docname, table_field) {
 	if(typeof n==typeof [])
 		refresh_many(n, docname, table_field);
 
-	if(table_field && cur_frm.fields_dict[table_field].grid.grid_rows_by_docname) { // for table
-		cur_frm.fields_dict[table_field].grid.grid_rows_by_docname[docname].refresh_field(n);
+	if (n && typeof n==='string' && table_field){
+		var grid = cur_frm.fields_dict[table_field].grid,
+			field = frappe.utils.filter_dict(grid.docfields, {fieldname: n});
+		if (field && field.length){
+			field = field[0];
+			var meta = frappe.meta.get_docfield(field.parent, field.fieldname, docname);
+			$.extend(field, meta);
+			if (docname){
```

Comment:
There I'm extending the field by the meta, because I discovered that `frappe.meta.get_docfield(doctype, fieldname, [docname])` dont return the docfield of the grid-form, the docfield is fetched in the line 71,